src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/impl/DeleteOperation.java

```diff
@@ -68,7 +71,34 @@ public Uni<Supplier<CommandResult>> execute(QueryExecutor queryExecutor) {
             .concatenate()
             .onItem()
             .transformToUniAndConcatenate(
-                readDocument -> deleteDocument(queryExecutor, delete, readDocument))
+                readDocument -> {
+                  AtomicInteger attempt = new AtomicInteger(0);
+                  return Multi.createBy()
+                      .repeating()
+                      .uni(() -> deleteDocument(queryExecutor, delete, readDocument, attempt))
+                      .whilst(
+                          respVal ->
+                              (respVal == DeleteResponse.CONCURRENCY_FAILURE
+                                  && attempt.incrementAndGet() < retryLimit))
+                      .collect()
+                      .last()
+                      .onItem()
+                      .transform(
+                          respVal -> {
+                            switch (respVal) {
+                              case DELETED:
+                                return true;
+                              case MODIFIED_BY_CONCURRENT_TRANSACTION:
+                                return false;
+                              case CONCURRENCY_FAILURE:
+                              default:
+                                throw new JsonApiException(
+                                    ErrorCode.CONCURRENCY_FAILURE,
+                                    "Delete failed for document with id %s because of concurrent transaction"
+                                        .formatted(readDocument.id().toString()));
+                            }
+                          });
```

Comment:
this should not be done like this, retries should be done with the Multiny support.. No need to have your own counter and everything, see: https://smallrye.io/smallrye-mutiny/2.1.0/tutorials/retrying/

Reply:
Did the retry logic in a previous commit before using onFailure().retry(), then changed it this logic because this will be easier to capture errors if case we want to return both deleted count and error document ids.
Will change it back to onFailure().retry().